File: src/main/java/dev/slne/surf/essentials/listeners/DeathListener.java
package dev.slne.surf.essentials.listeners;

import dev.slne.surf.essentials.utils.EssentialsUtil;
import dev.slne.surf.essentials.utils.color.Colors;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.PlayerDeathEvent;

public class DeathListener implements Listener {

    @EventHandler(priority = EventPriority.LOWEST)
    public void onPlayerDeath(PlayerDeathEvent event) {
        var message = event.deathMessage();
        if (message == null) return;
        message = message.replaceText(b -> b.matchLiteral(event.getPlayer().getName())
                .replacement(event.getPlayer().displayName()
                        .colorIfAbsent(Colors.TERTIARY)))
                .colorIfAbsent(Colors.INFO);

        event.deathMessage(EssentialsUtil.getPrefix()
                .append(message));
    }
}

File: src/main/kotlin/dev/slne/surf/essentials/listener/DeathListener.kt
package dev.slne.surf.essentials.listener

import dev.slne.surf.surfapi.core.api.messages.Colors
import dev.slne.surf.surfapi.core.api.messages.adventure.buildText
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.PlayerDeathEvent

object DeathListener : Listener {
    @EventHandler
    fun onDeath(event: PlayerDeathEvent) {
        val message = event.deathMessage() ?: return

        event.deathMessage(buildText {
            appendPrefix()
            append(message.replaceText {
                it.matchLiteral(
                    LegacyComponentSerializer
                        .legacyAmpersand()
                        .serialize(event.entity.teamDisplayName())
                        .replace("§", "")
                )
                    .once()
                    .replacement(event.entity.displayName())
            }.colorIfAbsent(Colors.GRAY))
        })
    }
}